src/Core/Framework/Plugin/Command/PluginListCommand.php

```diff
@@ -36,22 +36,30 @@ protected function configure(): void
     {
         $this
             ->setDescription('Show a list of available plugins.')
-            ->addOption('filter', 'f', InputOption::VALUE_REQUIRED, 'Filter the plugin list to a given term');
+            ->addOption('filter', 'f', InputOption::VALUE_REQUIRED, 'Filter the plugin list to a given term')
+            ->addOption('format', null, InputOption::VALUE_REQUIRED, 'Choose table or JSON format to output to', 'table');
     }
 
     /**
      * {@inheritdoc}
      */
     protected function execute(InputInterface $input, OutputInterface $output): int
     {
+        $format = strtolower($input->getOption('format'));
+        $isTable = $format === 'table';
+        $isJson = $format === 'json';
         $io = new ShopwareStyle($input, $output);
-        $io->title('Shopware Plugin Service');
+        if ($isTable) {
+            $io->title('Shopware Plugin Service');
+        }
         $context = Context::createDefaultContext();
 
         $criteria = new Criteria();
         $filter = $input->getOption('filter');
         if ($filter) {
-            $io->comment(sprintf('Filtering for: %s', $filter));
+            if ($isTable) {
```

Comment:
You could create two methods one for json and one for table. So you can cleanup the if else stuff on each output

Reply:
Done